---
title: "Technical comment on 'Negative-assortative mating for color in wolves'"
output:
  bookdown::pdf_document2:
    keep_tex: yes
    latex_engine: pdflatex
    number_sections: no
    toc: no
    includes:
      in_header: header.tex
bibliography: disassortative-mating.bib
link-citations: yes
linkcolor: blue
csl: evolution.csl
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
knitr::opts_knit$set(root.dir = "..")
```

```{r}
source("r/header.R")
library(magrittr)
pkg_info = sessioninfo::package_info(pkgs = c("reticulate", "rmarkdown")) |>
  dplyr::filter(package %in% c("reticulate", "rmarkdown")) %$%
  setNames(ondiskversion, package)
source_python('python/get-sympy-version.py')
py_info = py_discover_config(required_module = NULL, use_environment = NULL)
```

<!-- \begin{centering} -->
<!-- Christopher D. Muir\textsuperscript{1,*} -->

<!-- \bf{1} School of Life Sciences, University of Hawaii at M\=anoa, Honolulu, HI 96822, USA -->
<!-- \end{centering} -->

## Background

@hedrick_negative-assortative_2016 reported on "negative-assortative mating for color in wolves" from Yellowstone National Park, the "first documented case of significant negative-assortative mating in mammals." Based on the close correspondence of genotype and allele frequencies observed in the wild to that predicted by their population genetic model, they conclude that "negative-assortative mating could be entirely responsible for the maintenance of this well-known color polymorphism." While researching examples of nonrandom mating in the wild to teach in class I discovered that the results of their population genetic model are inconsistent with their stated assumptions, as I understand them. In this paper, I revisit the model with the following two objectives:

1. Demonstrate that the frequency of negative-assortative mating between gray and black pelage color morphs in their model does not follow from their assumptions; and
2. Derive results that are consistent with their assumptions.

I am critiquing only their model, not the data analysis or conclusions. Both the original model and the new model analyzed here lead to similar inferences about the maintenance of the pelage color polymorphism because the equilibrium genotype and allele frequencies are nearly the same in both models. However, it is important that the mathematical biology literature provide logically consistent analysis so that future researchers may benefit most from its insights.

## The frequency of assortative mating is inconsistent with the assumptions

@hedrick_negative-assortative_2016 assume that a proportion $A$ matings are assortative, but the proportion they derive is much less (see Fig. \@ref(fig:sample-space) for a graphical derivation). For consistency, I use the same symbols as @hedrick_negative-assortative_2016 (Table \@ref(tab:symbols) lists all symbols and their definitions). I infer three key assumptions from the two statements on the bottom-left of pg. 758 of @hedrick_negative-assortative_2016:

> "gray wolves have a genotype of $\mathit{kk}$ and an assumed frequency of $P$ and black
wolves have genotypes $\mathit{Kk}$ and $\mathit{KK}$ with frequencies $H$ and $Q$, respectively $(P + H + Q = 1)$." 

> "Assume that $A$ and $1-A$ are the proportions of negative-assortative mating and random mating, respectively, in the population."

From these statements, I infer that:

1. $\mathit{kk}$, $\mathit{Kk}$, and $\mathit{KK}$ are mutually exclusive genotypes with frequencies $P$, $H$, and $Q$
2. Negative-assortative mating and random mating are mutually exclusive mating types with frequencies $A$ and $1-A$
3. Genotype and mating type are independent ($\mathrm{Pr}[A \cap B] = \mathrm{Pr}[A] \times \mathrm{Pr}[B]$)

Other model assumptions made by @hedrick_negative-assortative_2016 are common in population genetics. The model is deterministic, individuals are monoecious, generations are discrete and nonoverlapping, and individuals can mate multiply. Some of these assumptions may not be valid for wolves and different assumptions could result in qualitatively different outcomes. For example, if individuals only mate in pairs, this would constrain the amount of negative assortative mating that is possible when either allele is rare (see below). While alternative assumptions may help understand how assortative mating maintains genetic variation, my purpose here is not to critique the assumptions of @hedrick_negative-assortative_2016, but rather to show their results are not logically consistent with their stated assumptions.

\begin{figure}[ht]
  \centering
  \includegraphics[width=\textwidth]{../figures/sample-space.pdf}
  \caption{(Caption next page.)}
  \label{fig:sample-space}
\end{figure}
\addtocounter{figure}{-1}

\begin{figure} [t!]
\caption{(Previous page.) A graphical summary of the sample space that illustrates why this model arrives at results that differ from Hedrick \textit{et al.} (2016). (a) The sample space consists of six mutually exclusive genotype-mating type combinations that must sum to 1 given the assumptions. (b) In this study we show that the the probabilities in the assortative mating subspace sum to $A$, as assumed, if one conditions on assortative mating consisting of a black and gray coat color parent. In contrast, Hedrick \textit{et al.} (2016) effectively assume that a proportion of assortatively mating individuals equal to the area of the gray boxes do not contribute matings. This is why the models reach different conclusions. (c) Both models treat randomly mating individuals identically.}
\end{figure}

```{r, symbols}

tribble(
  ~Symbol, ~`Variable string`, ~Description,
  "$k$", "$\\mathtt{k}$", "recessive beta-defensin variant",
  "$K$", "$\\mathtt{K}$", "dominant beta-defensin variant",
  "$p$", "$\\mathtt{p}$", "frequency of $\\textit{k}$ allele",
  "$q$", "$\\mathtt{q}$", "frequency of $\\textit{K}$ allele",
  "$P$", "$\\mathtt{P}$", "frequency of $\\textit{kk}$ genotype",
  "$H$", "$\\mathtt{H}$", "frequency of $\\textit{Kk}$ genotype",
  "$Q$", "$\\mathtt{Q}$", "frequency of $\\textit{KK}$ genotype",
  "$A$", "$\\mathtt{A}$", "proportion negative-assortatively mating"
) |>
  knitr::kable(format = "markdown", caption = "Glossary of mathematical symbols, variable string used in source code, and description.", align = "ccl")

```

Based on these assumptions, I deduce that:

1. The probability of all genotype-mating type combinations must sum to 1
2. The probability of all genotypes in the assortative-mating subspace must sum to $A$ 
3. The probability of all genotypes in the random-mating subspace must sum to $1-A$ 

There are six mutually exclusive genotype-mating type combinations in the population sample space (Fig. \ref{fig:sample-space}a). Since $P+H+Q=1$, $A + 1 - A = 1$, and genotype is independent of mating type, the probability of all genotype-mating type combinations must sum to 1.

\begin{multline*}
 \sum_i^{\{\mathit{kk}, \mathit{Kk}, \mathit{KK}\}} \sum_j^{\{\textrm{assort}, \textrm{random}\}} \textrm{Genotype}_i \cap \textrm{Mating~type}_j = \\  
  (\mathit{kk} \cap \textrm{assort}) + (\mathit{Kk} \cap \textrm{assort}) + 
  (\mathit{KK} \cap \textrm{assort}) + \\ (\mathit{kk} \cap \textrm{random}) + 
  (\mathit{Kk} \cap \textrm{random}) + (\mathit{KK} \cap \textrm{random}) \\
 = PA + HA + QA + P (1 - A) + H (1 - A) + Q (1 - A) \\
 = A (P + H + Q) + (1 - A) (P + H + Q) \\
 = 1 - A + A \\
 = 1 \\
\end{multline*}

Furthermore, we know that within the negative-assortative and random mating subspaces, the total probability must sum to $A$ and $1 - A$, respectively:

\begin{align*}
  A = & PA + HA + QA \\
  1 - A = & P (1 - A) + H (1 - A) + Q (1 - A)
\end{align*}
The model in @hedrick_negative-assortative_2016 is internally inconsistent because the proportion of negative-assortative matings does not equal $A$ as defined (Fig. \@ref(fig:sample-space)b).
$$A := \frac{\textrm{Assortative matings}}{\textrm{Total matings}} =\frac{\textrm{Assortative matings}}{\textrm{Assortative} + \textrm{Random matings}}$$
@hedrick_negative-assortative_2016 state that the frequency of assortative matings is $2 A P (H + Q)$ (*cf* top of pg. 759) and the frequency of random matings is $1 - A$. Applying these frequencies reveals that the proportion of assortative matings is not equal to $A$ as assumed:
$$ \frac{\textrm{Assortative matings}}{\textrm{Assortative} + \textrm{Random matings}} = \frac{2 A P (H + Q)}{2 A P (H + Q) + 1 - A}$$
There are no solutions to the expression above where the proportion of negative-assortative mating would equal $A$ when genotype and mating proportions are between 0 and 1. With their model, the actual proportion of negative-assortative would vary from 0 when $P = 0$ or $P = 1$ and $A / (2 - A)$ when $P = 0.5$.

To summarize, a proportion $A$ should mate assortatively given the assumptions of their model, but only $2 AP(H+Q)$ actually mate assortatively according to their results. In essence, they assign a proportion $A$ to mate assortatively, but then a proportion $A - 2 AP(H+Q)$ do not mate assortatively (the area of the gray regions in Fig. \@ref(fig:sample-space)b), and are therefore not counted among the total number of matings. This is why the probabilities of all matings do not sum to 1. The logically consistent solution is to condition on the fact that if a mating is negative-assortative it must by definition have one gray and one black parent (Fig. \@ref(fig:sample-space)b). In the next section, I use this approach to derive different results. In contrast, @hedrick_negative-assortative_2016 effectively impose selection without ever stating that assumption. To deal with the fact that resulting genotype frequencies do not sum to 1, they regularize the frequencies (*cf* equation 1a-b), as normally done in models of selection. Regularization is appropriate with selection because selection shrinks or expands the sample space as long as average fitness does not equal 1. @hedrick_negative-assortative_2016 impose selection because some of the assortative-mating individuals do not mate assortatively and are therefore not counted among the matings that result in offspring.

## Revised solutions consistent with model assumptions

The previous section showed that the frequency of assortative gray $\times$ black matings was not derived in a manner logically consistent with the model's assumptions. Here I derive new mating frequencies, genotype frequencies, and equilibria. This model was previously analyzed by @li_first_1976] where in his notation $D = H + Q$, $R = P$, and $w = A$ (*cf* pg. 367-368). I used Sympy version `r sympy_version` [@meurer_sympy:_2017] for symbolic derivations through Python version `r py_info[["version"]]` and the *R* package **reticulate** version `r pkg_info["reticulate"]` [@ushey_reticulate_2022]. All other computations were performed in *R* version `r getRversion()` [@r_core_team_r_2022]. 

Table \@ref(tab:probabilities) derives the probabilities of all possible outcomes and Table \@ref(tab:genotypes) summarizes the frequency each mating combination. This is the exact same process used to derive frequency of mating combinations in positive-assortative mating models [e.g. @hedrick_population_2012]. Since I model random mating identically to the previous model, the frequencies of gray $\times$ gray and black $\times$ black matings are identical; only the frequency of gray $\times$ black matings differs between models (Table \@ref(tab:genotypes); Fig. \@ref(fig:sample-space)c).

Code in the Supporting Information derives the expressions in (Table \@ref(tab:genotypes)) analytically using a computer algebra system, but one can also use the Law of Total Probability to prove it. The Law of Total Probability for discrete probability distributions states that $\mathrm{Pr}[A] = \sum_i \mathrm{Pr}[A|B_i] \mathrm{Pr}[B_i]$ where $\mathrm{Pr}[A|B_i]$ is the probability of outcome $A$ conditional on outcome $B_i$. The total probability of $A$ is the sum of conditional probabilities across all outcomes for event $B$. Using the Law of Total Probability, the probability of a gray $\times$ black mating is:
$$\mathrm{Pr}[\textrm{gray} \times \textrm{black}] = \mathrm{Pr}[\textrm{gray} \times \textrm{black}|\textrm{assort}] \mathrm{Pr}[\textrm{assort}] + \mathrm{Pr}[\textrm{gray} \times \textrm{black}|\textrm{random}] \mathrm{Pr}[\textrm{random}]$$
We already assume that $\mathrm{Pr}[\textrm{assort}] = A$ and $\mathrm{Pr}[\textrm{random}] = 1 - A$. With random mating, I arrive at the same expression as @hedrick_negative-assortative_2016, $\mathrm{Pr}[\textrm{gray} \times \textrm{black}|\textrm{random}] = 2P(H+Q)$ (*cf* top of pg. 759). If the mating is negative-assortative, then it *must* be a gray $\times$ black mating. Therefore, $\mathrm{Pr}[\textrm{gray} \times \textrm{black}|\textrm{assort}] = 1$. Putting these together, I obtain:

\begin{align}
  \mathrm{Pr}[\textrm{gray} \times \textrm{black}] = & 1 \times A + 2P(H+Q) \times (1-A) \nonumber \\
  = & A + 2 P (H + Q) (1 - A) \label{eq1}
\end{align}

This result diverges from that given in @hedrick_negative-assortative_2016, where they report the frequency of gray $\times$ black matings is $2 P (H + Q)$ (*cf* Table 1). One advantage of their result is that the frequency of assortative matings is 0 when the population is fixed at this locus ($P = 0$ or $H + Q = 0$). In contrast, Eq. \ref{eq1} conjures $A$ assortative matings from a population of wolves with identical coat color. This is a logical consequence of assuming that a proportion $A$ mate assortatively irrespective of the allele frequency. An alternative approach is to set bounds on $A$ as a function color morph frequency. For example, if we assume that wolves mate in monogamous pairs, then $A$ cannot exceed twice the frequency of the rarer color morph. Consider a single black wolf in pack of 100 ($P = 0.99$). The maximum value of $A$ is $1/50$ if there is one gray $\times$ black mating and 49 gray $\times$ gray matings. More generally, the maximum value of $A$ in a population of monogamous pairs can be found by solving Equation \ref{eq1} for $A$ when the maximum proportion of assortative matings is the lesser of $2P$ and $2(1 - P)$. When grey morphs are rarer than black morphs ($P < 0.5$), the maximum value of $A$ is $2P ^ 2 / (2P ^ 2 - 2 P + 1)$; when black morphs are rarer ($P > 0.5$), the maximum value of $A$ is $2 (P^2 - 2P + 1) / (2 P ^ 2 - 2 P + 1)$. Different constraints on $A$ apply to a population where individuals can self-fertilize and/or mate with multiple individuals. Incorporating constraints on $A$ as a function of phenotypic frequency may be important, but the assumptions stated by @hedrick_negative-assortative_2016 do not imply any constraint on $A$ when one color morph is rare.

```{r, probabilities}

read_rds("objects/table1_genotypes.rds") |>
  mutate(
    `Parent 1` = glue("${parent1}$"),
    `Pr[Parent 1]` = glue("${prob1_symbol}$"),
    `Pr[Mating]` = glue("${prob_mating_symbol}$"),
    `Parent 2` = glue("${parent2}$"),
    `Pr[Parent 2]` = glue("${prob2_symbol}$"),
    `Pr[Total]` = glue("${genotype_frequency_symbol}$") |>
      str_replace_all("\\*\\*", " ^ ") |> 
      str_replace_all("\\*", " ") 
  ) |>
  select(`Parent 1`, `Pr[Parent 1]`, Mating = mating, `Pr[Mating]`, `Parent 2`, 
         `Pr[Parent 2]`, `Pr[Total]`) |>
  arrange(desc(`Parent 1`), desc(`Parent 2`), Mating) |>
  mutate(Color = case_when(
    `Parent 1` == "$kk$" & `Parent 2` == "$kk$" ~ "Gray $\\times$ gray",
    `Parent 1` == "$kk$" & `Parent 2` == "$K-$" ~ "Gray $\\times$ black",
    `Parent 1` == "$K-$" & `Parent 2` == "$kk$" ~ "Gray $\\times$ black",
    `Parent 1` == "$K-$" & `Parent 2` == "$K-$" ~ "Black $\\times$ black",
  )) |>
  # Make notation more consistent:
  mutate(
    `Pr[Parent 1]` = str_replace(`Pr[Parent 1]`, "\\(1 - P\\)", "H + Q"),
    `Pr[Parent 2]` = str_replace(`Pr[Parent 2]`, "\\(1 - P\\)", "H + Q"),
    `Pr[Mating]` = str_replace(`Pr[Mating]`, "\\(1 - A\\)", "1 - A"),
    `Pr[Total]` = c("$0$", "$P^2 (1 - A)$", "$AP$", "$P (H + Q) (1 - A)$", 
                    "$A (H + Q)$", "$P (H + Q) (1 - A)$", "$0$", "$(H + Q)^2 (1 - A)$")
  ) |>
  knitr::kable(format = "markdown", caption = "The probability of every mating outcome in the negative-assortative mating model analyzed by Hedrick *et al.* (2016). For the notation, the probability of event *X* is Pr[*X*]. The total probabilities for each row are derived from the product of all probabilities in the same row, Pr[Total] = Pr[Parent 1] $\\times$ Pr[Mating] $\\times$ Pr[Parent 2].", align = "llllllll")

```

```{r, genotypes}

tribble(
  ~Color, ~`Mating Genotypes`, ~`Frequency (Hedrick *et al.* 2016)`, ~`Frequency (this paper)`,
  "Gray $\\times$ gray", "$kk \\times kk$", "$P^2 (1 - A)$", "$P^2 (1 - A)$",
  "Gray $\\times$ black", "$kk \\times K-$", "$2 P (H + Q)$", "$A + 2 P (H + Q) (1 - A)$",
  "Black $\\times$ black", "$K- \\times K-$", "$(H + Q) ^ 2 (1 - A)$", "$(H + Q) ^ 2 (1 - A)$"
) |>
  knitr::kable(format = "markdown", caption = "Hedrick *et al.* (2016) incorrectly derive the frequency of gray $\\times$ black. The corrected expressions are provided here.", align = "llll")

```

Despite the different frequency of gray $\times$ black matings resulting from each model, the equilibrium genotype frequencies are very similar. In both models, $\hat{P} = 0.5$, implying $0.5 = \hat{H} + \hat{Q}$. I find that $\hat{Q} = (A / 2 - \sqrt{2 (A + 1)} + 1.5)/(1 - A)$, which is close to the equilibrium values obtained in @hedrick_negative-assortative_2016 through recursion (Fig. \@ref(fig:fig2)). Next, I compared allele frequency change depicted in Figs. 3-4 of @hedrick_negative-assortative_2016 to that predicted with the new model. The effect of $A$ on change in the frequency of the $K$ allele is qualitatively similar, but much faster with the new model (Fig. \@ref(fig:fig3)). This is because the magnitude of allele frequency change far from the equilibrium is much greater with the new model (Fig. \@ref(fig:fig4)). As a result, @hedrick_negative-assortative_2016 overestimate how long it would take for $K$ to reach equilibrium given observed levels of assortative mating ($A = 0.430$). They conclude that $K$ would reach equilibrium at $\hat{q} = 0.278$ in 25 generations with $A = 0.430$. In the revised model, $K$ would reach equilibrium at $\hat{q} = 0.271$ in only 15 generations with $A = 0.430$ and a starting allele frequency $q_0 = 0.01$. Hence, the revised model actually lends credence to their conclusion that negative-assortative mating may be a better explanation than heterozygote advantage for variation at the beta definsin locus.

```{r fig2, fig.align='center', fig.width=6, fig.height=4, fig.cap="The equilibirum frequency of $Q$, the $\\mathit{KK}$ homozygote in this study (dashed line) and Hedrick \\textit{et al.} (2016) (solid line) for possible values of $A$, the proportion of wolves mating assortatively by color."}
df_fig = read_rds("objects/fig2.rds")

ggplot(df_fig, aes(A, Q, linetype = Source)) +
  geom_line(size = 1.2) +
  xlab(expression(italic(A)~(proportion~assortative~mating))) +
  ylab(expression(italic(Q)~(frequency~of~italic(KK)))) +
  theme_cowplot()


```

\begin{figure}[ht]
  \centering
  \includegraphics[width=\textwidth]{../figures/fig3.pdf}
  \caption{A comparison of the change in frequency of the black allele $K$ between Hedrick \textit{et al.} (2016) and this study. $K$ starts at a frequency 0.01 and the plots show the dynamics for three different levels of negative-assortative mating $A$.}
  \label{fig:fig3}
\end{figure}

\begin{figure}[ht]
  \centering
  \includegraphics[width=\textwidth]{../figures/fig4.pdf}
  \caption{A comparison of the change in frequency of the black allele $K$ between Hedrick \textit{et al.} (2016) and this study. The initial frequency of $K$ is along the $x$-axis and the change in $K$ along the $y$-axis where $A = 0.430$ as in Hedrick \textit{et al.} (2016) Fig. 4. The models have the similar equilibrua where they cross the gray dotted line at 0, but in this study the magnitude of change in allele frequency increases the system gets further from this equilibrium.}
  \label{fig:fig4}
\end{figure}

In conclusion, the logical inconsistency in @hedrick_negative-assortative_2016 does not undermine their primary conclusion that negative-assortative mating by color may explain the distribution of genotype frequencies at the beta definsin locus in the Yellowstone population of wolves (*Canis lupus*). The new derivation here may prove useful to future research on negative-assortative mating.

# Data availability

The source code is available in a public GitHub repository (https://github.com/cdmuir/disassortative-mating) and is archived on Zenodo (https://doi.org/10.5281/zenodo.7003312).

# Acknowledgements

Kirk Lohmueller and two anonymous reviewers provided feedback and mathematical derviations that greatly improved this manuscript.

# Literature cited