---
title: "Technical comment on 'Negative-assortative mating for color in wolves'"
output:
  bookdown::pdf_document2:
    keep_tex: yes
    latex_engine: pdflatex
    number_sections: no
    toc: no
    includes:
      in_header: header.tex
bibliography: disassortative-mating.bib
link-citations: yes
linkcolor: blue
csl: evolution.csl
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
knitr::opts_knit$set(root.dir = "..")
```

```{r}
source("r/header.R")
library(magrittr)
pkg_info = sessioninfo::package_info(pkgs = c("reticulate", "rmarkdown")) |>
  dplyr::filter(package %in% c("reticulate", "rmarkdown")) %$%
  setNames(ondiskversion, package)
source_python('python/get-sympy-version.py')
py_info = py_discover_config(required_module = NULL, use_environment = NULL)
```

Hedrick *et al.* (2016) reported on "negative-assortative mating for color in wolves" from Yellowstone National Park, the "first documented case of significant negative-assortative mating in mammals." Based on the close correspondence of genotype and allele frequencies observed in the wild to that predicted by their population genetic model, they conclude that "negative-assortative mating could be entirely responsible for the maintenance of this well-known color polymorphism." While researching examples of nonrandom mating in the wild to teach in class I discovered a mistake in their model. The mathematical error does not substantially alter their inference because the equilibrium genotype and allele frequencies are similar in both. However, it is important that the mathematical biology literature provide correct and logically consistent analysis so that future researchers may benefit most from its insights.

```{r, symbols}

tribble(
  ~Symbol, ~`Variable string`, ~Description,
  "$k$", "$\\mathtt{k}$", "recessive beta-defensin variant",
  "$K$", "$\\mathtt{K}$", "dominant beta-defensin variant",
  "$p$", "$\\mathtt{p}$", "frequency of $k$ allele",
  "$q$", "$\\mathtt{q}$", "frequency of $K$ allele",
  "$P$", "$\\mathtt{P}$", "frequency of $kk$ genotype",
  "$H$", "$\\mathtt{H}$", "frequency of $Kk$ genotype",
  "$Q$", "$\\mathtt{Q}$", "frequency of $KK$ genotype",
  "$A$", "$\\mathtt{A}$", "proportion negative-assortatively mating"
) |>
  knitr::kable(format = "markdown", caption = "Glossary of mathematical symbols, variable string used in source code, and description.", align = "ccl")

```

For consistency, I use the same symbols as @hedrick_negative-assortative_2016 (Table \@ref(tab:symbols) lists all symbols and their definitions). I used Sympy version `r sympy_version` [@meurer_sympy:_2017] for symbolic derivations through Python version `r py_info[["version"]]` and the *R* package **reticulate** version `r pkg_info["reticulate"]` [@ushey_reticulate_2022]. All other computations were performed in *R* version `r getRversion()` [@r_core_team_r_2021]. I have provided my source code for review and will deposit my code in public GitHub and Zenodo repositories upon publication.

In @hedrick_negative-assortative_2016, the frequency of gray $\times$ black matings is incorrectly written as $2 P (H + Q)$ (*cf* Table 1). However, this value does not account for all possible outcomes that result in gray $\times$ black matings (Table \@ref(tab:probabilities)). As a result, the genotype frequencies (Table \@ref(tab:genotypes)) do not sum to 1 as they should if they account for all possible outcomes. Ironically, I found an analogous derivation in @hedrick_population_2012 for positive-assortative mating where the genotype frequencies do sum to 1 when ignoring other evolutionary forces such as selection. The correct expression derived by summing all ways gray $\times$ black matings can occur is provided in Table \@ref(tab:genotypes). The corrected genotype frequencies sum to 1 as expected (see code for analytical derivation in Supporting Information).

```{r, probabilities}

read_rds("objects/table1_genotypes.rds") |>
  mutate(
    `Parent 1` = glue("${parent1}$"),
    `Pr[Parent 1]` = glue("${prob1_symbol}$"),
    `Pr[Mating]` = glue("${prob_mating_symbol}$"),
    `Parent 2` = glue("${parent2}$"),
    `Pr[Parent 2]` = glue("${prob2_symbol}$"),
    `Pr[Total]` = glue("${genotype_frequency_symbol}$") |>
      str_replace_all("\\*\\*", " ^ ") |> 
      str_replace_all("\\*", " ") 
  ) |>
  select(`Parent 1`, `Pr[Parent 1]`, Mating = mating, `Pr[Mating]`, `Parent 2`, 
         `Pr[Parent 2]`, `Pr[Total]`) |>
  arrange(desc(`Parent 1`), desc(`Parent 2`), Mating) |>
  mutate(Color = case_when(
    `Parent 1` == "$kk$" & `Parent 2` == "$kk$" ~ "Gray $\\times$ gray",
    `Parent 1` == "$kk$" & `Parent 2` == "$K-$" ~ "Gray $\\times$ black",
    `Parent 1` == "$K-$" & `Parent 2` == "$kk$" ~ "Gray $\\times$ black",
    `Parent 1` == "$K-$" & `Parent 2` == "$K-$" ~ "Black $\\times$ black",
  )) |>
  knitr::kable(format = "markdown", caption = "The probability of every mating outcome in the negative-assortative mating model analyzed by Hedrick *et al.* (2016). For the notation, the probability of event *X* is Pr[*X*]. The total probabilities for each row are derived from the product of all probabilities in the same row, Pr[Total] = Pr[Parent 1] $\\times$ Pr[Mating] $\\times$ Pr[Parent 2].", align = "llllllll")

```

```{r, genotypes}

tribble(
  ~Color, ~`Mating Genotypes`, ~`Frequency (Hedrick *et al.* 2016)`, ~`Frequency (this paper)`,
  "Gray $\\times$ grey", "$kk \\times kk$", "$P^2 (1 - A)$", "$P^2 (1 - A)$",
  "Gray $\\times$ black", "$kk \\times K-$", "$2 A P (H + Q)$", "$A P - A (H + Q) - 2 P (1 - A) (H + Q)$",
  "Black $\\times$ black", "$K- \\times K-$", "$(H + Q) ^ 2 (1 - A)$", "$(H + Q) ^ 2 (1 - A)$"
) |>
  knitr::kable(format = "markdown", caption = "Hedrick *et al.* (2016) incorrectly derive the frequency of gray $\\times$ black. The corrected expressions are provided here.", align = "llll")

```

@hedrick_negative-assortative_2016 account for the fact that genotype frequencies do not sum for 1 by regularizing the frequencies (*cf* equation 1a-b), as normally done in models of selection. In effect, by not accounting for all possible outcomes, they are accidentally assuming a type of selection. However, the equilibrium genotype frequencies they derive are very similar to the correct equilibrium. In both models, $\hat{P} = 0.5$, implying $0.5 = \hat{H} + \hat{Q}$. I find that $\hat{Q} = (-A / 2 + \sqrt{2 (A + 1)} - 1.5)/(A - 1)$, which is close to the equilibrium values obtained in @hedrick_negative-assortative_2016 through recursion (Fig. \@ref(fig:fig1)).

```{r fig1, fig.align='center', fig.width=6, fig.height=4, fig.cap="The equilibirum frequency of $Q$, the $KK$ homozygote in this study (dashed line) and Hedrick \\textit{et al.} (2016) (solid line) for possible values of $A$, the proportion of wolves mating assortatively by color."}
df_fig = read_rds("objects/fig.rds")

ggplot(df_fig, aes(A, Q, linetype = Source)) +
  geom_line(size = 1.2) +
  xlab(expression(italic(A)~(proportion~assortative~mating))) +
  ylab(expression(italic(Q)~(frequency~of~italic(KK)))) +
  theme_cowplot()


```

In conclusion, the mathematical error in @hedrick_negative-assortative_2016 does not undermine their primary conclusion that negative-assortative mating by color may explain the distribution of genotype frequencies at the beta definsin locus in the Yellowstone population of wolves (*Canis lupus*). The derivation here may prove useful to future research on negative-assortative mating.

# Literature cited